♻️ Refactor : 카드셋을 17종으로 확장 및 enum 기반 구조 적용

bisor0627 opened the pull request:
## ✨ What’s this PR?
### 📌 관련 이슈 (Related Issue)
<!-- 해당 PR이 어떤 이슈를 해결하는지 연결해주세요 -->
- #216

### 🧶 주요 변경 내용 (Summary)
기존 7종의 제한된 cardSet을 전면 리팩토링하여, 전체 17종 카드에 대한 enum 기반의 카드 정의를 적용했습니다.  
향후 다양한 카드 타입에 따른 로직 분기, UI 연결 등을 위한 구조 기반을 마련한 작업입니다.

- `PathCardType` enum 정의 (총 17종)
- 각 카드 타입의 `directions`, `connect`, `symbol` 정보를 기반으로 한 `cardSet` 재구성
- 기존 문자열 하드코딩 방식 제거 및 의미 기반 설계 적용

---

### 📸 스크린샷 (Optional)
<!-- UI 작업의 경우, 구현한 화면을 첨부해주세요 -->
<img width="300" alt="CLI" src="https://github.com/user-attachments/assets/f9ec1ea4-5953-416a-a9e8-5a7357ef3868" >

### 🧪 테스트 / 검증 내역
<!-- 동작 확인 여부나 시나리오 테스트 내용을 간단히 써주세요 -->

- [x] CLI 및 APP 정상 빌드
- [x] CLI 동작시 기존 7개 카드의 경우 `Board.isPlacable()` 가능
- [ ] 새로 추가된 카드의 경우 `Board.isPlacable()`...


### 💬 기타 공유 사항

- 기존에는 단 7장의 카드만 정의되어 있어 테스트 케이스 확장 및 실제 룰 구현에 제약이 있었습니다.
- 전체 카드 타입을 명시적으로 정의함으로써 카드별 특징(연결 방향, 방해 여부 등)을 보다 명확하게 표현하고 확장 가능성을 확보합니다.

### 🙇🏻‍♀️ 리뷰 가이드 (선택)

- `Board.isPlacable()` 로직에서 버그가 있는 것으로 추측...됩니다!
  - 현재 구조에서는 **기존 7종 카드만 연결이 허용**되고, 새로 추가된 카드 타입들은 대부분 배치가 거부됩니다.
  - 이는 `isPlacable()` 내부 로직을 분석한 후에 수정을 진행해야한다고 판단했습니다.
  - 본 PR에서는 해당 로직에는 손대지 않았습니다.

bisor0627 committed:
♻️ 카드셋을 17종으로 확장

Why:
- 기존 7장의 카드셋으로는 게임 로직 테스트와 이미지 매핑 확장에 한계가 있어, 전체 카드 타입을 반영한 정규 카드셋 구성이 필요했습니다.

How:
- PathCardType 기반 17종 카드의 directions, connect, symbol을 포함한 카드셋으로 cardSet을 리팩토링하였습니다.

Tags: #card #deck #refactor #gameLogic

pngxlols commented on SaboteurKit/Sources/SaboteurKit/Models/Card.swift:
```diff
@@ -19,11 +26,26 @@ public struct Card: Sendable {
+    Card(directions: [true, true, true, true], connect: true, symbol: "┼"),       // trbl
+
+    // 방해 카드
+    Card(directions: [false, false, false, false], connect: false, symbol: "⨯"),  // lBlock
```
요 카드는 어디에도 설치할 수 없기에 존재의 이유가 없습니다! 삭제하면 될 듯합니다

bisor0627 replied:
그럼 lBlock을 제거하고 총 16종의 카드셋으로 하겠습니다!

pngxlols commented on SaboteurKit/Sources/SaboteurKit/Models/Card.swift:
```diff
@@ -19,11 +26,26 @@ public struct Card: Sendable {
-    Card(directions: [false, false, false, false], connect: false, symbol: "💣"), // 폭탄
-    Card(directions: [true, true, true, true], connect: false, symbol: "⦻"), // 전방, 방해
+    // 연결 가능한 길 카드
+    Card(directions: [true, true, true, false], connect: true, symbol: "┴"),      // t
```
directions의 기준은 [상, 우, 하, 좌] 입니다. 따라서 해당과 같은 directions를 갖는 카드는 ㅏ 이런 모양을 가지고 있어야합니다
아마 directions 표기 방법이 달라서 오류가 발생하는 듯 싶네요!
top부터 시계방향으로 흘러간다고 생각하시면 됩니당
isPlacable() 함수도 그 기준에 맞춰서 만들어져있어요

pngxlols reviewed (commented):
이제봤네용 ... 리뷰를 처음 달아봐서 ... 죄성합니다

bisor0627 committed:
♻️ 카드 방향 정의 및 불필요한 카드 제거

Why:
- 방향 기준(top → 시계방향)이 일관되지 않아 카드 로직에서 오류가 발생함
- lBlock 카드는 실제 게임 보드에 설치될 수 없는 구조로 존재할 필요가 없음

How:
- directions 기준을 [top, right, bottom, left]로 정리하고 잘못 정의된 카드 방향 수정
- 사용되지 않는 lBlock 카드 제거 및 cardSet 총 16종으로 조정
- symbol 명확성을 위해 일부 특수문자 조정

Tags: #refactor #card #direction #review-reflection

pngxlols reviewed (approved):
완벽해요! 수고 넘 많으셨습니당
